web/src/engine/websites/MangaLib.ts

```diff
@@ -1,46 +1,125 @@
-@Common.MangasMultiPageCSS('/manga-list?page={page}', 'div.media-card-wrap a.media-card', 1, 1, 0, MangaExtractor)
-@Common.ChaptersSinglePageJS(chapterScript, 500)
-@Common.PagesSinglePageJS(pageScript, 500)
-@Common.ImageAjax()
+type ImageServers = {
+    imageServers: ImageServer[]
+}
+
+type ImageServer = {
+    id: string,
+    url: string,
+    site_ids: number[]
+}
+
+type APIResult<T> = {
+    data: T,
+    meta: {
+        has_next_page: boolean,
+    }
+}
+
+type APIManga = {
+    id: number,
+    rus_name: string,
+    slug_url: string
+}
+
+type APIChapter = {
+    volume: string,
+    number: string,
+    name: string,
+    branches_count: number,
+    branches: {
+        id: number,
+        branch_id: number | null,
+        teams: {
+            name: string
+        }[]
+    }[]
+}
+
+type APIPages = {
+    pages: {
+        url: string
+    }[]
+}
+
+type ChapterID = {
+    branch_id: string,
+    number: string,
+    volume: string
+}
+
+@Common.ImageAjax(true)
 export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://api.mangalib.me/api/';
+    private readonly siteId = 1;
+    private imageServer: ImageServer = undefined;
+    private readonly mangasTaskPool = new TaskPool(1, new RateLimit(2, 1));
 
     public constructor() {
-        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Russian, Tags.Source.Aggregator);
+        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Russian, Tags.Source.Aggregator);
     }
 
     public override get Icon() {
         return icon;
     }
 
+    public override async Initialize(): Promise<void> {
+        const { data: { imageServers } } = await FetchJSON<APIResult<ImageServers>>(new Request(new URL('constants?fields[]=imageServers', this.apiUrl)));
+        this.imageServer = imageServers.find(server => server.id === 'main' && server.site_ids.includes(this.siteId));
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/ru/manga/[^/]`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaSlug = new URL(url).pathname.split('/').at(-1);
+        const { data: { slug_url, rus_name } } = await FetchJSON<APIResult<APIManga>>(new Request(new URL(`manga/${mangaSlug}`, this.apiUrl)));
+        return new Manga(this, provider, slug_url, rus_name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const url = new URL(`manga?page=${page}&site_id[]=${this.siteId}`, this.apiUrl);
+            const { data, meta: { has_next_page } } = await this.mangasTaskPool.Add(() => FetchJSON<APIResult<APIManga[]>>(new Request(url)), Priority.Low);
```

Comment:
When setting the concurrency for `mangasTaskPool` to anything different than `1`, it will not work as intended.
Maybe choose more appropriate name to describe the behavior e.g., `mangasSequentialTaskPool`, `mangasNonCurrentTaskPool`, ...